src/main/java/bowling/model/frame/FrameNumber.java

```diff
@@ -0,0 +1,53 @@
+package bowling.model.frame;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class FrameNumber {
+    private static final String FRAME_NUMBER_ERROR = "범위를 벗어난 프레임 수 입니다.";
+    private static final int PLACE_VALUE = 2;
+
+    public static final int MAX_FRAME_NUMBER = 10;
+    public static final int MIN_FRAME_NUMBER = 1;
+
+    private static final Map<Integer, FrameNumber> frameNumberCache;
+
+    private final int frameNumber;
+
+    static {
+        frameNumberCache = IntStream.rangeClosed(MIN_FRAME_NUMBER, MAX_FRAME_NUMBER)
+                .boxed()
+                .collect(Collectors.toMap(Function.identity(), FrameNumber::new));
+    }
+
+    private FrameNumber(int number) {
+        this.frameNumber = number;
+    }
+
+    public static FrameNumber from(int number) {
+        if (frameNumberCache.containsKey(number)) {
+            return frameNumberCache.get(number);
+        }
+        throw new IllegalArgumentException(FRAME_NUMBER_ERROR);
+    }
+
+    public FrameNumber next() {
+        return FrameNumber.from(frameNumber + 1);
+    }
+
+    public boolean beforeLast() {
+        return frameNumber == MAX_FRAME_NUMBER - 1;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%0" + PLACE_VALUE + "d", frameNumber);
+    }
+
+
+    public static int compare(FrameNumber o1, FrameNumber o2) {
+        return o1.frameNumber - o2.frameNumber;
+    }
```

Comment:
FrameNumber도 Comparable을 구현하면 어떨까요 :) ?
```suggestion
    @Override
    public int compareTo(final FrameNumber o) {
        return Integer.compare(this.frameNumber, o.frameNumber);
    }
```

Reply:
Java Comparable.compareTo(T o) 문서에서
(x.compareTo(y)==0) == (x.equals(y)) 구현하도록 강하게 권장하고 있습니다.

![image](https://user-images.githubusercontent.com/20432323/102063226-15849300-3e39-11eb-8673-433d8fcdceac.png)

> compareTo가 "같다" 라고 판단했다면, equals가 true여야 한다.

compareTo와 equals 의 결과가 같지 않다면, 컬렉션 규약을 위반할 수 있습니다.
실제로 Collections, Map 등에서 정렬된 컬렉션들의 동치성 비교에는 equals 대신 compareTo를 사용하기 때문입니다.

예로 Java 5에서 PriorityQueue.remove() 메소드는 compareTo ()에 의존했지만
Java 6부터 equals()에 의존했던 경우가 있었으며

HashSet에 담길 경우 equals 메소드로 비교하며, TreeSet에 담길 경우 compareTo 메소드로 비교하기 때문에
해당 객체가 어느 컬렉션에서 사용될지 모르기에 해당 규약이 위반되지 않게 조건을 준수해도록 수정해주시면 좋을 것 같아요!